lib/protocol/bep_extensions.go

```diff
@@ -252,11 +269,26 @@ func (f FileInfo) isEquivalent(other FileInfo, comp FileInfoComparison) bool {
 			}
 		}
 		if f.Platform.Windows != nil && other.Platform.Windows != nil {
-			if *f.Platform.Windows != *other.Platform.Windows {
+			if f.Platform.Windows.OwnerName != other.Platform.Windows.OwnerName ||
+				f.Platform.Windows.OwnerIsGroup != other.Platform.Windows.OwnerIsGroup {
 				return false
 			}
 		}
 	}
+	if !comp.IgnoreXattrs && f.Platform != other.Platform {
```

Comment:
Should this really check all OSes? It is used to e.g. compare an ad hoc create fileinfo from disk with one from a remote. If that one accumulated multiple OSes it will never be equal, would it?

Reply:
This outer `f.Platform != other.Platform` check is just a shortcut to catch the easy case where all platform values are equal or nil, then we don't need to do per-platform comparisons. If there is something unequal we'll do the inner `xattrsEqual` checks which return true if either side is nil (nothing to compare to). That is, things will only be considered *unequal* iff there is data for a given platform on both sides and it differs.

A fileinfo created on Linux will never be unequal to another just because the other has Windows attributes, for example -- it will only be unequal if there are also Linux attributes and those differ.